appservice/src/deploy/waitForDeploymentToComplete.ts

```diff
@@ -87,50 +96,51 @@ export async function waitForDeploymentToComplete(client: SiteClient, kuduClient
-
     if (permanentId) {
-            // Use "permanentId" to find the deployment during its "permanent" phase
-            deployment = await kuduClient.deployment.getResult(permanentId);
-        } else if (expectedId) {
-            // if we have a "expectedId" we know which deployment we are looking for, so wait until latest id reflects that
-            try {
-                const latestDeployment: DeployResult = await kuduClient.deployment.getResult('latest');
-                [deployment, permanentId] = latestDeployment.id === expectedId ? [latestDeployment, latestDeployment.id] : [undefined, undefined];
-            } catch (error) {
-                const parsedError: IParsedError = parseError(error);
-                // swallow 404 error since "latest" might not exist on the first deployment
-                if (parsedError.errorType !== '404') {
-                    throw parsedError;
-                }
-            }
-        } else if (initialStartTime) {
-            // Use "initialReceivedTime" to find the deployment during its "temp" phase
-            deployment = (await kuduClient.deployment.getDeployResults())
-                // tslint:disable-next-line:no-non-null-assertion
-                .filter((deployResult: DeployResult) => deployResult.startTime && deployResult.startTime >= initialStartTime!)
-                .sort((a: DeployResult, b: DeployResult) => nonNullProp(b, 'startTime').valueOf() - nonNullProp(a, 'startTime').valueOf())
-                .shift();
-            if (deployment && !deployment.isTemp) {
-                // Make note of the id once the deplyoment has shifted to the "permanent" phase, so that we can use that to find the deployment going forward
-                permanentId = deployment.id;
-            }
-        } else {
-            // Use "latest" to get the deployment before we know the "initialReceivedTime" or "permanentId"
-            try {
-                deployment = <DeployResult | undefined>await kuduClient.deployment.getResult('latest');
-            } catch (error) {
-                const parsedError: IParsedError = parseError(error);
-                // swallow 404 error since "latest" might not exist on the first deployment
-                if (parsedError.errorType !== '404') {
-                    throw parsedError;
-                }
+        // Use "permanentId" to find the deployment during its "permanent" phase
+        deployment = await kuduClient.deployment.getResult(permanentId);
+    } else if (expectedId) {
+        // if we have a "expectedId" we know which deployment we are looking for, so wait until latest id reflects that
+        try {
+            const latestDeployment: DeployResult = await kuduClient.deployment.getResult('latest');
+            // if the latest deployment is temp, then a deployment has triggered so we should watch it even if it doesn't match the expectedId
+            deployment = latestDeployment.isTemp ? latestDeployment : undefined;
```

Comment:
Isn't `deployment` ovewritten the next line?

Reply:
Only if there is a match.  The reason for this is if the latest is a temp, even if it doesn't match our current ID, we should track it because that means something has triggered.  Once it does match, we know that we can save it as our permanentId.

Reply:
> Only if there is a match

But if there's not a match, doesn't the next line set it to undefined?

I feel like you want to do something like this instead:
```
if (latestDeployment.isTemp) {
    deployment = latestDeployment;
} else if (latestDeployment.id === expectedId) {
    deployment = latestDeployment;
    permanentId = latestDeployment.id;
}
```